fendermint/vm/actor_interface/src/ipc.rs

```diff
@@ -28,6 +31,37 @@ pub mod gateway {
         pub majority_percentage: u8,
     }
 
+    impl TryFrom<GatewayParams> for ConstructorParameters {
+        type Error = fvm_shared::address::Error;
+
+        fn try_from(value: GatewayParams) -> Result<Self, Self::Error> {
+            let mut route = Vec::new();
+            for addr in value.subnet_id.children() {
+                let addr = match addr.payload() {
+                    Payload::ID(id) => EthAddress::from_id(*id),
+                    Payload::Delegated(da)
+                        if da.namespace() == eam::EAM_ACTOR_ID && da.subaddress().len() == 20 =>
+                    {
+                        EthAddress(da.subaddress().try_into().expect("checked length"))
+                    }
+                    _ => return Err(fvm_shared::address::Error::InvalidPayload),
+                };
+                route.push(H160::from(addr.0))
+            }
+            Ok(Self {
+                network_name: SubnetID {
+                    root: value.subnet_id.root_id(),
+                    route,
+                },
+                bottom_up_check_period: value.bottom_up_check_period,
+                top_down_check_period: value.top_down_check_period,
+                // XXX: Ignoring any error resulting from larger fee than what fits into U256. This is in genesis after all.
```

Comment:
Actually, should we panic here if this happens? As it is in genesis is not part of the core operation and it would allow us to catch potential misconfigurations.

Reply:
Thanks, added a test to show that it indeed panics.